plugin/build.gradle.kts

```diff
@@ -243,10 +231,85 @@ java {
   withJavadocJar()
 }
 
-tasks.shadowJar.configure {
-  archiveClassifier.set("")
+if(project.properties["rfg.skipJavadoc"].toString().toBoolean()) {
+  tasks.named("javadoc").configure { enabled = false }
+}
+
+val depsShadowJar = tasks.register<ShadowJar>("depsShadowJar") {
+  archiveClassifier.set("deps")
+  archiveVersion.set("0.0") // constant version to prevent task from rerunning when project version changes
   isEnableRelocation = true
   relocationPrefix = "com.gtnewhorizons.retrofuturagradle.shadow"
+  configurations.add(project.configurations.runtimeClasspath.get())
+  dependencies {
+    // we're already shading this in combinedShadowJar
+    exclude(project(":oldasmwrapper"))
+  }
+}
+
+val mainShadowJar = tasks.register<ShadowJar>("mainShadowJar") {
+  archiveClassifier.set("mainShadow")
+
+  val prefix = "com.gtnewhorizons.retrofuturagradle.shadow"
+
+  from(sourceSets.main.get().output)
+
+  // Adapted from Shadow's RelocationUtil.groovy
+  // We want to relocate references to dependencies but not include the dependencies themselves in this jar.
+  // We also don't want to double-shade RFG classes
+  val packages = mutableSetOf<String>()
+
+  val configurations = listOf(project.configurations.runtimeClasspath.get())
+
+  configurations.iterator().forEach { configuration ->
```

Comment:
Can this be wrapped in a doFirst{} block? otherwise you're forcing early dependency resolution when the task objects are constructed

Reply:
It probably can, that would also better match how Shadow does it (it only calls `RelocationUtil` as the first step of executing the copy task).